code/parse/sexp.cpp

```diff
@@ -17367,6 +17369,59 @@ int sexp_goal_incomplete(int n)
 		return SEXP_TRUE;
 }
 
+/**
+ * Resets an event, its status, and its nodes so that it is as if the event had never been evaluated.
+ */
+void sexp_reset_event(int node)
+{
+	for (int n = node; n >= 0; n = CDR(n))
+	{
+		auto name = CTEXT(n);
+		int event_num = mission_event_lookup(name);
+		if (event_num >= 0)
+		{
+			auto eventp = &Mission_events[event_num];
+
+			eventp->result = 0;
+			eventp->timestamp = TIMESTAMP::invalid();
+
+			// clear some flags but not all
```

Comment:
I don't know if this comment adds any value. It just says what the code does, and there isn't any complex magic here that benefits from being summarized. The comment would be more useful if it said why it's clearing these flags in particular.

Reply:
Good point